### PR TITLE
[CHK-1594] feat(wallet-fe): add wallet-fe CDN

### PR DESCRIPTION
**Pull request by giovanniberti**
### List of changes

 * add wallet-fe CDN

### Motivation and context

Added CDN to host web content for the new eCommerce wallet web views :)

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

Depends on #1180

---

### CODE REVIEW

**Review by ciuffagianluca — DISMISSED** [review]
Well done!

File: src/domains/wallet-common/05_wallet_cdn.tf [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    # HSTS
+    modify_response_header_action = [{
+      action = "Overwrite"
+      name   = "Strict-Transport-Security"
+      value  = "max-age=31536000"
+      },
+      # Content-Security-Policy
+      {
+        action = "Overwrite"
+        name   = var.env_short == "p" ? "Content-Security-Policy-Report-Only" : "Content-Security-Policy"
```
Comment (pietro-tota) [inline src/domains/wallet-common/05_wallet_cdn.tf:50]: Why `Content-Security-Policy-Report-Only` for prod env? I would set `Content-Security-Policy` as this product is a new one. What do you think?

File: src/domains/wallet-common/99_variables.tf [hunk 2]
```diff
@@ -83,18 +83,38 @@ variable "ingress_load_balancer_ip" {
...
+
 variable "dns_zone_internal_prefix" {
   type        = string
   default     = null
   description = "The dns subdomain."
 }
 
+variable "dns_default_ttl_sec" {
+  type        = number
+  description = "value"
```
Comment (pietro-tota) [inline src/domains/wallet-common/99_variables.tf:114]:
```suggestion
  description = "The DNS default TTL in seconds"
```

File: src/domains/wallet-common/00_dns_public.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  tags = var.tags
+}
+
+//Prod ONLY record to DEV public DNS delegation
+resource "azurerm_dns_ns_record" "dev_wallet" {
+  count               = var.env_short == "p" ? 1 : 0
+  name                = "dev"
+  zone_name           = azurerm_dns_zone.wallet_public.name
+  resource_group_name = data.azurerm_resource_group.rg_vnet.name
+  records = [
```
Comment (infantesimone) [inline src/domains/wallet-common/00_dns_public.tf:14]: TODO update after delegation

**Comment by infantesimone** [comment]
Add availability test like https://github.com/pagopa/pagopa-infra/blob/96620fe82290cc3f17ae4e547b140e35c8072a78/src/core/monitor.tf#L129 useful for verifying the certificate expiration

File: src/domains/wallet-common/env/weu-dev/terraform.tfvars [hunk 4]
```diff
@@ -24,8 +24,12 @@ log_analytics_workspace_resource_group_name = "pagopa-d-monitor-rg"
 
 ingress_load_balancer_ip = "10.1.100.250"
 
+## DNS
+
 external_domain          = "pagopa.it"
 dns_zone_internal_prefix = "internal.dev.platform"
+dns_zone_prefix          = "dev.wallet"
```
Comment (infantesimone) [inline src/domains/wallet-common/env/weu-dev/terraform.tfvars:31]: To update with `payment-wallet`

File: src/domains/wallet-common/env/weu-uat/terraform.tfvars [hunk 5]
```diff
@@ -24,8 +24,12 @@ log_analytics_workspace_resource_group_name = "pagopa-u-monitor-rg"
 
 ingress_load_balancer_ip = "10.1.100.250"
 
+## DNS
+
 external_domain          = "pagopa.it"
 dns_zone_internal_prefix = "internal.uat.platform"
+dns_zone_prefix          = "uat.wallet"
```
Comment (infantesimone) [inline src/domains/wallet-common/env/weu-uat/terraform.tfvars:31]: ditto

**Review by infantesimone — CHANGES_REQUESTED** [review]

**Comment by infantesimone** [comment]
you need to add the configurations for prod to be able to require dns delegation https://pagopa.atlassian.net/wiki/spaces/DEVOPS/pages/426214700/Azure+-+DNS+deleghe+sotto+domini+per+prodotto#2.-Creare-la-dns-zone-product.pagopa.it

File: src/domains/wallet-common/05_wallet_cdn.tf (hunk 1 above)
Comment (giovanniberti) [reply src/domains/wallet-common/05_wallet_cdn.tf:50]: I left it as it is on the Checkout CDN for now given that their stack is similar

File: src/domains/wallet-common/env/weu-dev/terraform.tfvars (hunk 4 above)
Comment (giovanniberti) [reply src/domains/wallet-common/env/weu-dev/terraform.tfvars]: Fixed in 9f2a4bf6792e5fa01f309595f1e445a690468fdd

File: src/domains/wallet-common/env/weu-uat/terraform.tfvars (hunk 5 above)
Comment (giovanniberti) [reply src/domains/wallet-common/env/weu-uat/terraform.tfvars]: Fixed in 9f2a4bf6792e5fa01f309595f1e445a690468fdd

File: src/domains/wallet-common/00_dns_public.tf (hunk 3 above)
Comment (giovanniberti) [reply src/domains/wallet-common/00_dns_public.tf:14]: Fixed in 922eb311

**Review by infantesimone — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**